addons/pkg/util/addon_util.go

```diff
@@ -218,3 +229,51 @@ func IsPackageInstallPresent(ctx context.Context,
 
 	return true, nil
 }
+
+// AddFinalizerToCR adds finalizer to the custom resource if not present and
+// returns true if finalizer is added
+func AddFinalizerToCR(log logr.Logger, addonName string, addonCR client.Object) bool {
+	// add finalizer to addon custom resource
+	if !controllerutil.ContainsFinalizer(addonCR, addontypes.AddonFinalizer) {
+		log.Info("Adding finalizer to addon custom resource", constants.AddonNameLogKey, addonName)
+		controllerutil.AddFinalizer(addonCR, addontypes.AddonFinalizer)
+		return true
+	}
+
+	return false
+}
+
+// RemoveFinalizerFromCR removes finalizer from the custom resource if present and
+// returns true if finalizer is removed
+func RemoveFinalizerFromCR(log logr.Logger, addonName string, addonCR client.Object) bool {
+	// remove finalizer from addon custom resource
+	if controllerutil.ContainsFinalizer(addonCR, addontypes.AddonFinalizer) {
+		log.Info("Removing finalizer from addon custom resource", constants.AddonNameLogKey, addonName)
+		controllerutil.RemoveFinalizer(addonCR, addontypes.AddonFinalizer)
+		return true
+	}
+
+	return false
+}
+
+// get paths for external CRDs by introspecting versions of the go dependencies
+func GetExternalCRDPaths(externalDeps map[string][]string) ([]string, error) {
```

Comment:
This is a test util function, I don't think we should put it here.

Reply:
Agreed, moved it to testutil directory instead